stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/RabbitMQTransactionContext.java

```
@@ -0,0 +1,137 @@
+    private Channel channel;
+    private final String connectorId;
+    private long deliveryTag = -1;
+    private boolean autoAck = true;
+
+
+    /**
+     * Sets the value of autoAck.
+     *
+     * @param autoAck Ack mode of the consumer.
+     */
+    public void setAutoAck(boolean autoAck) {
+        this.autoAck = autoAck;
+    }
+
+    /**
+     * Sets the delivery tag of the message.
+     *
+     * @param deliveryTag Delivery tag of the message.
+     */
+    public void setDeliveryTag(long deliveryTag) {
+        this.deliveryTag = deliveryTag;
+    }
+
+    /**
+     * Initializes the rabbitmq transaction context.
+     *
+     * @param channelObject RabbitMQ Channel object.
+     * @param connectorId   Connector Id.
+     */
+    public RabbitMQTransactionContext(BMap<String, BValue> channelObject, String connectorId) {
+        channel = (Channel) channelObject.getNativeData(RabbitMQConstants.CHANNEL_NATIVE_OBJECT);
+        this.connectorId = connectorId;
+    }
+
+    @Override
+    public void commit() {
+        try {
+            if (deliveryTag != -1 && !autoAck) {
+                channel.basicAck(deliveryTag, false);
+            }
+            channel.txCommit();
+        } catch (IOException exception) {
+            throw new RabbitMQConnectorException(RabbitMQConstants.COMMIT_FAILED
+                    + exception.getMessage(), exception);
+        }
+    }
+
+    @Override
+    public void rollback() {
+        try {
+            if (deliveryTag != -1 && !autoAck) {
+                channel.basicNack(deliveryTag, false, true);
+            }
+            channel.txRollback();
+        } catch (IOException exception) {
+            throw new RabbitMQConnectorException(RabbitMQConstants.ROLLBACK_FAILED
+                    + exception.getMessage(), exception);
+        }
+
+    }
+
+    @Override
+    public void close() {
+        // Do nothing
+    }
+
+    @Override
+    public XAResource getXAResource() {
+        // Not supported
```

We should return a runtime exception here saying not supported.

---

Resolved